bin/test_with_coverage.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:convert' show utf8, LineSplitter;
+
+import 'package:args/args.dart';
+import 'package:package_config/package_config.dart';
+import 'package:path/path.dart' as path;
+import 'package:coverage/src/util.dart' show extractVMServiceUri;
+
+final allProcesses = <Process>[];
+
+Future<void> dartRun(List<String> args,
+    {Function(String)? onStdout, String? workingDir}) async {
```

Comment:
[nit] avoid the implicit `dynamic` in the return type.

```suggestion
    {void Function(String)? onStdout, String? workingDir}) async {
```